roles/job/tasks/main.yml

```diff
@@ -38,6 +38,8 @@
   when:
     - ansiblejob_info['resources'][0]['status']['isFinished'] is defined
     - ansiblejob_info['resources'][0]['status']['isFinished']
+    - ansiblejob_info['resources'][0]['status']['generation'] is defined
+    - ansiblejob_info['resources'][0]['status']['generation']|int == ansiblejob_info['resources'][0]['metadata']['generation']|int
```

Comment:
For ansiblejob's and workflows we might only want to allow one reconciliation for.... Since we'd want users to be intentional about launching a job.  And a change to job_tags or something like that would trigger another job run, perhaps unintentionally.

Reply:
I agree with you @rooftopcellist the AnsibleJob is only meant to run only once.
The AnsibleJob CR is meant to represent an instance of a Job that is kicked off in AWX/AAP so you always have a corresponding AnsibleJob CR with the Job-ID/url of the actual job in AWX.
If we allow re-run then it will generate a new job-ID/url so we will have to make some additional changes to make sure we don't lose the previously ran job ID in the AnsibleJob status.